xarray/coding/variables.py

```diff
@@ -574,3 +575,34 @@ def decode(self, variable: Variable, name: T_Name = None) -> Variable:
             return variable
         else:
             return variable
+
+
+class EnumCoder(VariableCoder):
+    """Encode and decode Enum to CF"""
+
+    def encode(self, variable: Variable, name: T_Name = None) -> Variable:
+        """From python Enum to CF"""
+        dims, data, attrs, encoding = unpack_for_encoding(variable)
+        if attrs.get("enum"):
```

Comment:
Do only encode if this is `Enum`. If we did not decode in the first place this will fail.

Reply:
@bzah It looks like `.decode` isn't used any more. With the current 

```python
attrs["enum"] = Enum(var.datatype.name, var.datatype.enum_dict)
```
the netCDF4.EnumType is translated to Python enum on read and visa versa for write.

AFAICS `decode_enum=True/False` doesn't do anything. I'll add a review comment .